### PR TITLE
suggestion for fix for issue #2461

### PR DESCRIPTION
**Pull request by cameron-simpson**
As described in issue #2461, the `SentryWrappingMiddleware` MRO is just `object` if Django < 3.1 (when async middleware became a thing), but the `async_capable` check inside the class only looks for the `async_capable` attribute inside the middleware class.

This PR makes that check also conditional on Django >= 3.1.

Otherwise the code calls `super(.....).__init__(get_response)` and for Django < 3.1 this only finds `object.__init__`, not the wrapped middleware `__init__`.

Fixes GH-2461

### CODE REVIEW

File: sentry_sdk/integrations/django/middleware.py [hunk 1]
```diff
@@ -34,10 +34,11 @@
     import_string_name = "import_string"
 
 
-if DJANGO_VERSION < (3, 1):
-    _asgi_middleware_mixin_factory = lambda _: object
-else:
+django_supports_async_middleware = DJANGO_VERSION >= (3, 1)
```
Comment (szokeasaurusrex) [inline sentry_sdk/integrations/django/middleware.py:37]: Please use all caps variable name for constants like this one, i.e. `DJANGO_SUPPORTS_ASYNC_MIDDLEWARE`

**Review by szokeasaurusrex — CHANGES_REQUESTED** [review]
Thank you for the PR! I have left one comment on your proposed changes

Additionally, it looks like some of our required CI checks are failing with this pull request -- if you can, please try to debug these. For the linter check, you should make sure to run the file you changed through our code formatter, `black` (I believe this happens automatically if you have the pre-commit hooks configured). There is also a test failure; please try to solve this if you can; otherwise, if you struggle to figure it out, let us know, and we can help.

File: sentry_sdk/integrations/django/middleware.py (hunk 1 above)
Comment (cameron-simpson) [reply sentry_sdk/integrations/django/middleware.py]: I've applied `black` and adjusted the variable case.

I'm puzzling over the CI failure here:
https://github.com/getsentry/sentry-python/actions/runs/6660942801/job/18340882553

It looks like the version of pytest is too old for upcoming versions of pytest-asyncio. However, I don't see how this should happen only on my branch and not also on the main sentry codebase. I've merged recent commits from master into my branch, but none seems relevant to this test so I don't imagine the next CI run should be any better. But please push the button on it just for the exercise.

I'll keep looking a bit at this today. Is there a guide to running these tests locally?

**Comment by szokeasaurusrex** [comment]
We have our [contributing docs here](https://github.com/getsentry/sentry-python/blob/master/CONTRIBUTING.md), which contains some information about running our tests, but unfortunately the guide is not super detailed. If you can figure it out, I would suggest running the test suite with `tox`, which will allow you to run the tests with the same configuration as they are run in CI (in terms of Python version and installed dependencies). Otherwise, let me know and I will try to see if I can debug the test failures.

Also, now that the code formatting was fixed, I can see in our [linting CI check](https://github.com/getsentry/sentry-python/actions/runs/6750780566/job/18391738135?pr=2466) that there is a typing issue. Please try to fix this as well. We use the `mypy` linter, which you should also be able to run locally.

**Comment by szokeasaurusrex** [comment]
@cameron-simpson It actually seems that you can ignore the failing AWS Lambda tests for now; we recently opened an issue (#2487) because these tests have been failing for all contributor PRs.

**Comment by szokeasaurusrex** [comment]
Hey @cameron-simpson, we fixed the issue with our AWS Lambda tests. They currently are failing, but that is because for security reasons, I need to add a label to your PR to indicate that I have verified that your code won't leak our AWS Lambda secrets before the tests are allowed to run. Since I need to add this label every time you update the code, I will add this label once all other non-AWS Lambda checks are passing. 

It looks like our CI checks and Python 3.8 common tests are failing. Please fix these, and then I will allow the AWS Lambda tests to run

**Review by sentrivana — APPROVED** [review]

**Review by szokeasaurusrex — APPROVED** [review]

**Comment by sentrivana** [comment]
We can merge this as soon as master is green

**Pull request merged**